utils/benchmark.py

```diff
@@ -64,16 +65,34 @@ def parse_args():
     parser = ArgumentParser()
     parser.add_argument('-c', '--content-type', type=str, required=True,
                         help='Content Type of the input file')
-    parser.add_argument('-f', '--filename', type=str, required=True,
+    parser.add_argument('-f', '--filename', type=str, required=False,
                         help='File to be uploaded')
+    parser.add_argument('--data-size', metavar='SIZE',
+                        type=int, required=False,
+                        help='Size of generated data' +
+                        ' to be used instead of real file')
     return parser.parse_args()
 
 
+def get_random_bytes(size, seed):
+    random.seed(seed)
+    return random.bytes(size)
+
+
+def open_data(args):
+    if args.filename is not None:
+        return open(args.filename, 'rb')
+    if args.data_size is not None:
+        return BytesIO(get_random_bytes(args.data_size, 42))
+    raise Exception('Not enough arguments passed: ' +
+                    'please specify --filename or --data_size argument')
+
+
 @c_profile()
 def main():
     args = parse_args()
 
-    with open(args.filename, 'rb') as fd:
+    with open_data(args) as fd:
```

Comment:
This actually makes the `-f` option useless, since even if a file is passed using `-f`, the script generates random data anyway. Could you add a `--mode` parameter which allows you to either use a given file or use random bytes?

Also, after this change, what's the difference between the `utils/benchmark.py` script and the `utils/speedtest.py` script?

Reply:
Ah, somehow I failed to notice that. Never mind, looks good!

I would still prefer to keep `benchmark,py`. It has been useful in the past.